ops/charms/node_base.py

```diff
@@ -41,21 +48,34 @@ class NodeLabelError(Exception):
 
     def __init__(
         self,
-        charm: Charm,
+        charm: CharmImpl,
         kubeconfig_path: Union[PathLike, str],
+        *,
         kubectl: Optional[PathLike] = "/snap/bin/kubectl",
-    ):
+        user_label_key: str = "labels",
+        timeout: Optional[PositiveInt] = None,
+    ) -> None:
+        """Initialize the LabelMaker.
+
+        @param charm: The charm instance
+        @param kubeconfig_path: Path to the kubeconfig file
+        @param kubectl: Path to the kubectl binary
+        @param user_label_key: The key in the charm config where the user labels are stored
+        @param timeout: Number of seconds to retry a command
+        """
+
         super().__init__(parent=charm, key="NodeBase")
         self.charm = charm
         self.kubeconfig_path = kubeconfig_path
         self.kubectl_path = kubectl
+        self.user_labels_key = user_label_key
+        self.run_retries = RUN_RETRIES if timeout is None else timeout
```

Comment:
`RUN_RETRIES` sounds more like a number of attempts rather than a timeout. Could we rename the constant to something clearer, like `DEFAULT_TIMEOUT`?

Reply:
Done...